cmd/plume/prerelease.go

```diff
@@ -0,0 +1,233 @@
+	client, err := auth.GoogleClient()
+	if err != nil {
+		return err
+	}
+
+	src, err := storage.NewBucket(client, spec.SourceURL())
+	if err != nil {
+		return err
+	}
+
+	// Sanity check!
+	if vertxt := src.Object(src.Prefix() + "version.txt"); vertxt == nil {
+		verurl := src.URL().String() + "version.txt"
+		plog.Fatalf("File not found: %s", verurl)
+	}
+
+	plog.Printf("Running Azure pre-release...")
+
+	if err := azurePreRelease(ctx, client, src, &spec); err != nil {
+		return err
+	}
+
+	plog.Printf("Pre-release complete, run `plume release` to finish.")
+
+	return nil
+}
+
+// bunzip2 unzips file in into file out
+func bunzip2(in, out string) error {
+	inf, err := os.Open(in)
+	if err != nil {
+		return err
+	}
+
+	defer inf.Close()
+
+	bzr := bzip2.NewReader(inf)
+
+	outf, err := os.OpenFile(out, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return err
+	}
+
+	defer outf.Close()
+
+	_, err = io.Copy(outf, bzr)
+	return err
+}
+
+// azurePreRelease runs everything necessary to prepare a CoreOS release for Azure.
+//
+// This includes uploading the vhd image to Azure storage, creating an OS image from it,
+// and replicating that OS image.
+func azurePreRelease(ctx context.Context, client *http.Client, src *storage.Bucket, spec *channelSpec) error {
+	// construct azure api client
+	plog.Printf("Creating Azure API from %q", azureOpts.PublishSettingsFile)
+	a, err := azure.New(&azureOpts)
+	if err != nil {
+		return fmt.Errorf("failed to create Azure API: %v", err)
+	}
+
+	plog.Printf("Fetching Azure storage credentials")
+
+	storageKey, err := a.GetStorageServiceKeys(spec.Azure.StorageAccount)
+	if err != nil {
+		return err
+	}
+
+	// download azure vhd image and unzip it, but only if it doesn't exist
+	cachedir := filepath.Join(sdk.RepoCache(), "images")
+	vhdfile := filepath.Join(cachedir, spec.Azure.Image)
+	unzipped := strings.TrimSuffix(vhdfile, filepath.Ext(vhdfile))
+
+	if _, err := os.Stat(unzipped); err == nil {
+		plog.Printf("Reusing existing image %q", unzipped)
+	} else {
+		vhduri, err := url.Parse(spec.Azure.Image)
+		if err != nil {
+			return err
+		}
+
+		vhduri = src.URL().ResolveReference(vhduri)
+
+		plog.Printf("Downloading Azure image %q to %q", vhduri, vhdfile)
+
+		if err := sdk.UpdateFile(vhdfile, vhduri.String(), client); err != nil {
+			return err
+		}
+
+		// decompress it
+		plog.Printf("Decompressing %q...", vhdfile)
+
+		unzipped := strings.TrimSuffix(vhdfile, filepath.Ext(vhdfile))
+		if err := bunzip2(vhdfile, unzipped); err != nil {
+			return err
+		}
+	}
+
+	// sanity check - validate VHD file
+	plog.Printf("Validating VHD file %q", unzipped)
+	if err := validator.ValidateVhd(unzipped); err != nil {
+		return err
+	}
+
+	if err := validator.ValidateVhdSize(unzipped); err != nil {
+		return err
+	}
+
+	// upload blob, do not overwrite
+	plog.Printf("Uploading %q to Azure Storage...", unzipped)
+
+	blobName := fmt.Sprintf("coreos-%s-%s.vhd", specVersion, specChannel)
```

Comment:
from https://github.com/coreos/mantle/blob/master/cmd/plume/specs.go#L160, effectively they are just what you give plume pre-release as arguments.